libafl_frida/src/asan/hook_funcs.rs

```diff
@@ -825,15 +825,25 @@ impl AsanRuntime {
         _file_offset_low: u32,
         size: usize,
     ) -> *const c_void {
-        let original: extern "C" fn(*const c_void, u32, u32, u32, usize) -> *const c_void =
-            unsafe { std::mem::transmute(self.hooks.get(&"MapViewOfFile".to_string()).unwrap().0) };
-        let ret = (original)(
-            _handle,
-            _desired_access,
-            _file_offset_high,
-            _file_offset_low,
-            size,
-        );
+        extern "system" {
```

Comment:
This should now use the new static mut directly, instead of link time linkage.

Reply:
It is difficult to do this as the static mut is in register_hooks. So, I added an alternative hook macro called `hook_func_nolinkage` where it passes the original function pointer to the hook. I'm not sure if there is a better way to do this.